examples/move/scratch_off/sources/game.move

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module scratch_off::game {
```

Comment:
Could you add a description here explaining what this example is demonstrating, and what patterns it's using?